Bottom: ae5098c56bd72d71ba895c2d103aa694117906bc
Top:    e03171a78d0eb28f932718ca1f58bca5de67a53f
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-16 09:17:44 -0500

Enable file-based secrets provider


---
diff --git a/roles/buildbot-master/files/master.cfg b/roles/buildbot-master/files/master.cfg
index 11942aaa..c453a09b 100644
--- a/roles/buildbot-master/files/master.cfg
+++ b/roles/buildbot-master/files/master.cfg
@@ -5,7 +5,7 @@
 # 'master.cfg' in your buildmaster's base directory.
 
 import os
-from buildbot.plugins import steps, util
+from buildbot.plugins import secrets, steps, util
 
 os.environ['LONG_RUN_TIMEOUT'] = '14400'
 
@@ -22,6 +22,9 @@ c['change_source'] = []
 # --master option)
 c['protocols'] = {"pb": {"port": int(os.getenv('BUILDBOT_WORKER_PORT', 9989))}}
 
+# SECRETS MANAGEMENT
+c['secretsProviders'] = [secrets.SecretInAFile(dirname="/run/secrets")]
+
 # STATUS TARGETS
 
 # 'status' is a list of Status Targets. The results of each build will be
diff --git a/roles/buildbot-worker-ntel/tasks/main.yml b/roles/buildbot-worker-ntel/tasks/main.yml
index c6c57101..7c865e5c 100644
--- a/roles/buildbot-worker-ntel/tasks/main.yml
+++ b/roles/buildbot-worker-ntel/tasks/main.yml
@@ -79,6 +79,8 @@
   copy:
     content: "{{ known_hosts }}\n"
     dest: '{{ secrets_hostdir }}/{{ secrets_known_hosts }}'
+    group: '{{ buildbot_uid }}'
+    owner: '{{ buildbot_uid }}'
 
 - name: Checkout image repo
   git:
diff --git a/roles/buildbot-worker-ptxdist/tasks/main.yml b/roles/buildbot-worker-ptxdist/tasks/main.yml
index 46bf6355..c3a63545 100644
--- a/roles/buildbot-worker-ptxdist/tasks/main.yml
+++ b/roles/buildbot-worker-ptxdist/tasks/main.yml
@@ -86,6 +86,8 @@
   copy:
     content: "{{ known_hosts }}\n"
     dest: '{{ secrets_hostdir }}/{{ secrets_known_hosts }}'
+    group: '{{ buildbot_uid }}'
+    owner: '{{ buildbot_uid }}'
 
 - name: Checkout image repo
   git:
